airbyte-commons-worker/src/main/java/io/airbyte/workers/internal/DefaultAirbyteStreamFactory.java

```diff
@@ -61,7 +91,24 @@ public Stream<AirbyteMessage> create(final BufferedReader bufferedReader) {
     final var metricClient = MetricClientFactory.getMetricClient();
     return bufferedReader
         .lines()
-        .peek(str -> metricClient.distribution(OssMetricsRegistry.JSON_STRING_LENGTH, str.length()))
+        .peek(str -> metricClient.distribution(OssMetricsRegistry.JSON_STRING_LENGTH, str.getBytes(StandardCharsets.UTF_8).length))
+        .peek(str -> {
+          if (exceptionClass.isPresent()) {
+            final long messageSize = str.getBytes(StandardCharsets.UTF_8).length;
```

Comment:
at this point, isn't `str` already occupying as many as `str.getBytes(StandardCharsets.UTF_8).length` bytes in memory?